dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java

```diff
@@ -322,15 +325,52 @@ private void createTabAndSetBehaviour(final Session newSession) {
 
         });
         treeTab.setOnCloseRequest(SafeHandler.logHandle(event -> {
-            ds3SessionStore.removeSession(getSession());
-            closeTab((Tab) event.getSource(), getSession());
+            final ImmutableList<Ds3JobTask> notCachedRunningTasks = getActiveItemsInSession();
+            if (Guard.isNullOrEmpty(notCachedRunningTasks)) {
+                ds3SessionStore.removeSession(getSession());
+                closeTab((Tab) event.getSource(), getSession());
+            } else {
+                ds3Alert.showConfirmationAlert(resourceBundle.getString("confirmation"),
+                        notCachedRunningTasks.size() + " " + resourceBundle.getString("stillInProgress"),
+                        Alert.AlertType.CONFIRMATION,
+                        null,
+                        resourceBundle.getString("cancelJobsAndCloseTab"),
+                        resourceBundle.getString("returnToBrowser"))
+                        .ifPresent(response -> {
+                            if (response.equals(ButtonType.OK)) {
+                                ds3SessionStore.removeSession(getSession());
+                                //notCachedRunningTasks.forEach(ds3JobTask -> ds3JobTask.cancel(true));
+                                closeTab((Tab) event.getSource(), getSession());
+                            } else if (response.equals(ButtonType.CANCEL)) {
+                                event.consume();
+                            } else {
+                                LOG.error("Got a button click that shold not be possible");
```

Comment:
I know this is probably never going to be hit, but for future debugging would it be useful to say what the button type is that was theoretically not possible?

Reply:
Makes sense, I'll add the button type that we got sent.